src/inmanta/env.py

```diff
@@ -1428,7 +1429,7 @@ def exists(self) -> bool:
         """
         Returns True iff the venv exists on disk.
         """
-        return os.path.exists(self.python_path) and os.path.exists(self._path_pth_file)
```

Comment:
only used on the code path I'm working on.

I suspect the second check was there to force pth file creation, this restores the semantics to be in-line with the name.

Reply:
This was a mechanism to make sure that we recover if a venv creation failed half way. The creation of the pth file is the last step in the venv creation process. If that file exists, we can be pretty sure that the venv was created successfully. But maybe we have to replace this with a marker file. That would make the code more clear and it's also more correct since the creation of the marker file is an atomic operation. The creation of the pth file isn't.

Reply:
Mhh, how did we previously handle broken venv's on this code path I wonder?

Reply:
we would fall into this code path:

```python
        if not os.path.exists(self._path_pth_file):
        if not os.path.exists(self._path_pth_file):
            # Venv was created using an older version of Inmanta -> Update pip binary and set sitecustomize.py file
            # Venv was created using an older version of Inmanta -> Update pip binary and set sitecustomize.py file
            self._write_pip_binary()
            self._write_pip_binary()
            self._write_pth_file()
```

I wonder how to best handle this. 

I was now (for backward compat) considering this to be something the user has to fix.

Reply:
> I was now (for backward compat) considering this to be something the user has to fix.

Fine by me.

Reply:
Yes, but it also requires it to work (which implies many, many more things), the pth file is a very particular addition